### DISCUSSION TITLE
Is there an real examples of calling functions from the demos? Is there documentation how to make an XML graph editor?

### DISCUSSION THREAD

**Discussion by abramenko95** [maxGraph/maxGraph#596] [Q&A] [answered]
I need to make an editor that will get a diagram in XML as input.
I tried to use the mxGraph editor example but couldn't resolve all the dependencies:
https://github.com/jgraph/mxgraph/blob/master/javascript/examples/grapheditor/www/index.html
I tried to use maxGraph demos but I couldn't understand where some functions, arguments import from:
https://maxgraph.github.io/maxGraph/demo/?path=/story/xml-json-fileio--default
https://maxgraph.github.io/maxGraph/demo/?path=/story/editing-editing--default
How can I use this demos in my Vue.js project? Can you get some advices for resolve my task?

**Comment by tbouffard** [comment, marked as answer]
Hi, the graph editor examples has not been migrated from mxGraph. It currently lives in https://github.com/maxGraph/maxGraph/tree/v0.14.0/packages/html/stashed/grapheditor and hasn't been touched. There is no plan to migrate them for now (see other discussions and issues if you want more details about past decisions/discussions)
So it is not a good starting point, at least for now.

For the "demos", the UI provides a tab including some source code, but as you noticed, it miss imports and other things.
The source code is available in the repository:
- https://maxgraph.github.io/maxGraph/demo/?path=/story/xml-json-fileio--default --> https://github.com/maxGraph/maxGraph/blob/v0.14.0/packages/html/stories/FileIO.stories.ts
- https://maxgraph.github.io/maxGraph/demo/?path=/story/editing-editing--default --> https://github.com/maxGraph/maxGraph/blob/v0.14.0/packages/html/stories/Editing.stories.js

In addition, the `js-example` package in this repository is importing an maxGraph model stored as XML string. This is a vanilla JS project built with webpack.
So, it should contain everything you need for your use case. See https://github.com/maxGraph/maxGraph/blob/v0.14.0/packages/js-example/src/index.js

To finish this answer, I started working on a PR to add a dedicated page in the documentation about this feature: https://github.com/maxGraph/maxGraph/pull/524. If you see specific elements that should be included in such documentation, please share them with us.

Please let me know if this unblocks your problem. 😄

**Reply by abramenko95** [reply]
Thank you very much! This is a great start for my project! Now I have a problem with adding the custom object palette. Any ideas on how to do this?
https://github.com/maxGraph/maxGraph/discussions/600
